pkg/controller/job/job_controller_test.go

```diff
@@ -904,14 +924,16 @@ func TestControllerSyncJob(t *testing.T) {
 	}
 }
 
-func checkIndexedJobPods(t *testing.T, control *controller.FakePodControl, wantIndexes sets.Int, jobName string) {
+func checkIndexedJobPods(t *testing.T, control *controller.FakePodControl, wantIndexes, doNotWantIndexes sets.Int, jobName string) {
 	t.Helper()
 	gotIndexes := sets.NewInt()
 	for _, p := range control.Templates {
```

Comment:
`control.Templates` only has objects that the controller created. It has no information about deleted pods, so the new code is not adding any new coverage.

I would leave it out. Otherwise, you can validate `fakePodControl.DeletePodName`

Reply:
Ah ok, good to know. Done.